src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/InvokeTypeInfo.cs

```diff
@@ -77,8 +77,8 @@ public override void WriteData(PropertyValue value)
         {
             if (this.properties != null)
             {
-                var membersNames = new List<string>();
-                var membersValues = new List<object?>();
+                var membersNames = new List<string>(this.properties.Length);
+                var membersValues = new List<object?>(this.properties.Length);
```

Comment:
Do these even need to be lists?  From a quick look at EventPayload, it seems these could just as easily be arrays?

Reply:
That's a good point.  Yes, they can just be arrays.  I've pushed an update.